programs/golangci-lint.nix

```
@@ -0,0 +1,25 @@
+{ lib, pkgs, config, ... }:
+let
+  cfg = config.programs.golangci-lint;
+in
+{
+  options.programs.golangci-lint = {
+    enable = lib.mkEnableOption "golangci-lint";
+    package = lib.mkPackageOption pkgs "golangci-lint" { };
+    goPackage = lib.mkPackageOption pkgs "go" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    settings.formatter.golangci-lint = {
+      command = pkgs.writeShellApplication {
+        name = "golangci-lint-run";
+        runtimeInputs = [
+          cfg.package
+          cfg.goPackage
+        ];
+        text = "${cfg.package}/bin/golangci-lint run";
```

the program should take a list of arguments, otherwise, it will re-format the whole project every time.
```suggestion
        text = "${cfg.package}/bin/golangci-lint run \"$@\"";
```

---

Thanks! I applied the suggestion.